src/components/File.vue

```diff
@@ -21,108 +21,203 @@
  -->
 
 <template>
-	<a :class="{
-			'file--cropped': croppedLayout,
-		}"
-		class="file"
-		:href="davPath"
-		:aria-label="ariaLabel"
-		@click.prevent="openViewer">
-		<div v-if="item.injected.mime.includes('video') && item.injected.hasPreview" class="icon-video-white" />
-		<!-- image and loading placeholder -->
-		<transition-group name="fade" class="transition-group">
-			<img v-if="!error"
-				ref="img"
-				:key="`${item.injected.basename}-img`"
-				:src="src"
-				:alt="item.injected.basename"
-				:aria-describedby="ariaUuid"
-				@load="onLoad"
-				@error="onError">
-
-			<svg v-if="!loaded || error"
-				:key="`${item.injected.basename}-svg`"
-				xmlns="http://www.w3.org/2000/svg"
-				viewBox="0 0 32 32"
-				fill="url(#placeholder__gradient)">
-				<use v-if="isImage" href="#placeholder--img" />
-				<use v-else href="#placeholder--video" />
-			</svg>
-		</transition-group>
-
-		<!-- image name and cover -->
-		<p :id="ariaUuid" class="hidden-visually">{{ item.injected.basename }}</p>
-		<div class="cover" role="none" />
-	</a>
+	<div class="file-container"
+		:class="{selected}">
+		<CheckboxRadioSwitch v-if="allowSelection"
```

Comment:
Accessibility: checkbox without visible or invisble label

Reply:
Unsure what the solution is. Something like that ?

```suggestion
		<CheckboxRadioSwitch v-if="allowSelection"
			class="selection-checkbox"
			:checked="selected"
			@update:checked="onToggle">
			<span style="visibility: hidden">{{ t('photos', 'Select image {imageName}', {imageName: item.basename}) }}</span>
		</CheckboxRadioSwitch>
```

Reply:
I used the above solution, so marking as resolve, feel free to ping me if you were thinking of something else :)

Reply:
<this should use a `<label>` with a `for="<id of the checkboxradio`"` or a `aria-labelledby=""` (but make sure it also appear in the final `<input>` element as I'm not sure that `CheckboxRadioSwitch` propagate the `$args`)

Reply:
Can't it with your suggestion, as `CheckboxRadioSwitch` does not propagate `$args` and the `id` is unknown. Above solution works :).

Reply:
it doesn't really work since the screen reader needs the label to map a checkbox to a description, otherwise the checkbox and description are not linked

Probably will need additional API in nextcloud-vue

Reply:
The content passed to `CheckboxRadioSwitch` is inside a `<label>` isn't it enought ?

Reply:
Should be enough yes, the `CheckboxRadioSwitch` is already accessible